polus-imagej-plugins/polus-imagej-image-integral-plugin/src/main.py

```diff
@@ -0,0 +1,164 @@
+'''
+This file is autogenerated from an ImageJ plugin generation pipeline.
+'''
+
+from bfio.bfio import BioReader, BioWriter
+import argparse, logging
+import numpy as np
+from pathlib import Path
+import ij_converter
+import jpype, imagej, scyjava
+import typing
+
+# Initialize the logger
+logging.basicConfig(format='%(asctime)s - %(name)-8s - %(levelname)-8s - %(message)s',
+                    datefmt='%d-%b-%y %H:%M:%S')
+logger = logging.getLogger("main")
+logger.setLevel(logging.INFO)
+
+def main(_opName: str,
+         _in1: Path,
+         _out: Path,
+         ) -> None:
+             
+    """ Initialize ImageJ """ 
+    # Bioformats throws a debug message, disable the loci debugger to mute it
+    def disable_loci_logs():
+        DebugTools = scyjava.jimport("loci.common.DebugTools")
+        DebugTools.setRootLevel("WARN")
+    scyjava.when_jvm_starts(disable_loci_logs)
+    
+    # This is the version of ImageJ pre-downloaded into the docker container
+    logger.info('Starting ImageJ...')
+    ij = imagej.init("sc.fiji:fiji:2.1.1+net.imagej:imagej-legacy:0.37.4",
+                     headless=True)
+    ij_converter.ij = ij
+    logger.info('Loaded ImageJ version: {}'.format(ij.getVersion()))
+    
+    """ Validate and organize the inputs """
+    args = []
+    arg_types = []
+    arg_len = 0
+
+    
+    # Validate opName
+    opName_values = [
+        "DefaultIntegralImg",
+        "SquareIntegralImg",
+        "WrappedIntegralImg",
+    ]
+    assert _opName in opName_values, 'opName must be one of {}'.format(opName_values)
+    
+    # Validate in1
+    in1_types = { 
+        "DefaultIntegralImg": "RandomAccessibleInterval",
+        "SquareIntegralImg": "RandomAccessibleInterval",
+        "WrappedIntegralImg": "RandomAccessibleInterval",
+    }
+    
+    # Check that all inputs are specified 
+    if _in1 == None and _opName in list(in1_types.keys()):
+        raise ValueError('{} must be defined to run {}.'.format('in1',_opName))
```

Comment:
The purpose of this check is because different ops require different inputs. For example, one ImageJ op might require `_in`, while another operation that this plugin handles might not.